alternator/streams.cc

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <type_traits>
+#include <boost/lexical_cast.hpp>
+
+#include "base64.hh"
+#include "log.hh"
+#include "database.hh"
+
+#include "cdc/log.hh"
+#include "cdc/generation.hh"
+#include "db/system_distributed_keyspace.hh"
+#include "utils/UUID_gen.hh"
+
+#include "executor.hh"
+#include "tags_extension.hh"
+#include "rmw_operation.hh"
+
+template<typename ValueType, typename T>
+struct from_string_helper {
+    static bool Is(const ValueType& v) {
+        try {
+            Get(v);
+            return true;
+        } catch (...) {
+            return false;
+        }
+    }
+    static T Get(const ValueType& v) {
```

Comment:
No, the get previously was to enable overloaded conversion of any type supported by rapidjson conversion types. This declares a type helper to make marshalling the types used in stream.cc easy. 
I opted against putting this in rjson.hh for now, since it would add some template bloat. Eventually, if some other file also would like nice, automatic object marshalling, it should be moved there.

Reply:
Could we perhaps move all this automatic marshalling stuff out of both rjson.hh and from here into one single header file rjson_marshalling.hh or whatever, and then it will be clearer what all this stuff does, and which of the source code files use these tricks (streams.cc) and which don't (at least not yet)?